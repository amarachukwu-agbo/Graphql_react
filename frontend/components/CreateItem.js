import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import Error from './ErrorMessage';
import Router from 'next/router';

export const CREATE_ITEM_MUTATION = gql`
  mutation CREATE_ITEM_MUTATION(
    $title: String!
    $description: String!
    $price: Int!
    $image: String
    $largeImage: String) {
    createItem(
      title: $title
      description: $description
      price: $price
      image: $image
      largeImage: $largeImage
    ) {
      id
    }
  }
`;
export default class CreateItem extends Component {
  state = {
    title: '',
    description: '',
    image: '',
    largeImage: '',
    price: 0
  }

  handleChange = (e) => {
    const { name, value, type } = e.target;
    const val = type === 'number' ? parseFloat(value) : value;
    this.setState({
      [name]: val
    });
  }

  uploadImage = async (e) => {
    const { files } = e.target;
    const data = new FormData();
    data.append('file', files[0]);
    data.append('upload_preset', 'sickfits');
    const res = await fetch('https://api.cloudinary.com/v1_1/ama-hello-books-v2/image/upload', {
      method: 'POST',
      mode: 'cors',
      body: data
    });
    const file = await res.json();
    this.setState({
      image: file.secure_url,
      largeImage: file.eager[0].secure_url
    });
  }
  render() {
    const { title, price, description, image } = this.state;
    return (
      <Mutation mutation={CREATE_ITEM_MUTATION} variables={this.state}>
      {(createItem, { loading, error}) => (
        <Form onSubmit={async (e) => {
          e.preventDefault();
          const res = await createItem();
          Router.push({
            pathname: '/item',
            query: { id: res.data.createItem.id}
          });
        }}>
          <h2> Sell an item</h2>
          <Error error={error} />
          <fieldset disabled={loading} aria-busy={loading}>
          <label htmlFor="file">
            File
            </label>
            <input
              type="file"
              id="file"
              name="file"
              placeholder="Upload an image"
              required
              onChange={this.uploadImage}
            />
            {
              image && <img src={image} alt="uploaded image" width="50px"/>
            }
            <label htmlFor="title">
            Title
            </label>
            <input
              type="text"
              id="title"
              name="title"
              placeholder="Title"
              required
              value={title}
              onChange={this.handleChange}
            />

            <label htmlFor="price">
            Price
            </label>
            <input
              type="number"
              id="price"
              name="price"
              placeholder="Price"
              required
              value={price}
              onChange={this.handleChange}
            />

            <label htmlFor="description">
            Description
            </label>
            <textarea
              type="text"
              id="description"
              name="description"
              placeholder="Enter a description"
              required
              value={description}
              onChange={this.handleChange}
            />
            <button type="submit">Submit</button>
          </fieldset>
        </Form>
      )}
      </Mutation>
    )
  }
}
