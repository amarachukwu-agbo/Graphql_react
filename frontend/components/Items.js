import React, { Component } from 'react';
import { Query } from 'react-apollo';
import gql from 'graphql-tag';
import styled from 'styled-components';
import Item from './Item';
import Pagination from './Pagination';
import { perPage } from '../config';


export const ITEMS_QUERY = gql`
  query ITEMS_QUERY($skip: Int = 0, $first: Int = ${perPage}){
    items(skip: $skip, first: $first) {
      id
      title
      price
      description
      image
      largeImage
    }
  }
`;

const Center = styled.div`
  text-align: center;
`;

const ItemsList = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 60px;
  margin: 0 auto;
  max-width: ${props => props.theme.maxWidth};
`;

export default class Items extends Component {
  render() {
    const { page } = this.props;
    return (
      <Center>
      <Query query={ITEMS_QUERY} variables={{
        skip: (page - 1) * perPage
      }}>
        {({ error, loading, data }) => {
          if (loading) return <p>Loading... </p>
          if (error) return <p>{`Error: ${error.message}`}</p>
          console.log(data, 'data =====');
          return data.items.length ? (
            <div>
              <Pagination page={page}/>
              <ItemsList>
                {
                  data.items.map(item => <Item key ={item.id} item={item}/>)
                }
              </ItemsList>
              <Pagination page={page}/>
            </div>
          ) : <p>No items have been created :(</p>
        }}
      </Query>
      </Center>
    )
  }
}
