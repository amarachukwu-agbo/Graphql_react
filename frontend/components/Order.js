import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { format } from 'date-fns';
import gql from 'graphql-tag';
import Head from 'next/head';
import OrderStyles from './styles/OrderStyles';
import Error from '../components/ErrorMessage';
import formatMoney from '../lib/formatMoney';

export const SINGLE_ORDER_QUERY = gql`
  query SINGLE_ORDER_QUERY($id: ID!) {
    order(id: $id) {
      total
      charge
      items {
        id
        quantity
        price
        title
        image
      }
    }
  }
`;


export default class Order extends Component {
  static propTypes = {
    orderId: PropTypes.string.isRequired
  }
  render() {
    const { orderId } = this.props;
    return (
      <Query query={SINGLE_ORDER_QUERY} variables={{id: orderId}}>
        {({ data: {order}, loading, error }) => {
          if (error) return <Error error ={error} />
          if (loading) return <p>Loading ....</p>
          return (
            <OrderStyles>
              <Head>
                <title>Sick Fits - Order {orderId}</title>
              </Head>
              <p>
                <span>Order ID: </span>
                <span>{ orderId }</span>
              </p>
              <p>
                <span>Charge: </span>
                <span>{ order.charge }</span>
              </p>
              <p>
                <span>Date: </span>
                <span>{ format(order.createdAt, 'MMMM d, YYYY h:mm a') }</span>
              </p>
              <p>
                <span>Order Total: </span>
                <span>{ formatMoney(order.total) }</span>
              </p>
              <p>
                <span>Item Count: </span>
                <span>{ order.items.length }</span>
              </p>
              <div className="items">
                {
                  order.items.map(item => (
                    <div className="order-item" key={item.id}>
                      <img src={item.image} alt={item.title}/>
                      <div className="item-details">
                        <h2>{item.title}</h2>
                        <p>Quantity: {item.quantity}</p>
                        <p>Each: {formatMoney(item.price)}</p>
                        <p>Subtotal: {formatMoney(item.quantity * item.price)}</p>
                      </div>
                    </div>
                  ))
                }
              </div>
            </OrderStyles>
          )
        }}
      </Query>
    )
  }
}
