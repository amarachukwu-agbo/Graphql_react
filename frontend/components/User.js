import React from 'react'
import PropTypes from 'prop-types'
import gql from 'graphql-tag';
import { Query } from 'react-apollo';

export const CURRENT_USER_QUERY = gql`
  query CURRENT_USER_QUERY {
    me {
      id
      email
      name
      permissions
      orders
      cart {
        id
        quantity
        item {
          id
          price
          description
          image
          title
        }
      }
    }
  }

`;
const User = props => (
  <Query query={CURRENT_USER_QUERY}>
    {(payload) => {
      return props.children(payload)
    }}
  </Query>
);

User.propTypes = {
  children: PropTypes.func.isRequired
}

export default User;

