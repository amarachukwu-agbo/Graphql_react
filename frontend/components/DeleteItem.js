import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { ITEMS_QUERY } from './Items';

const DELETE_ITEM_MUTATION = gql`
  mutation DELETE_ITEM_MUTATION($id: ID!) {
    deleteItem(id: $id) {
      id
      title
    }
  }
`;

export default class DeleteItem extends Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired
  }
  update(cache, payload) {
    const { items } = cache.readQuery({ query: ITEMS_QUERY });
    const requiredItems = items
      .filter(item => item.id !== payload.data.deleteItem.id)
    cache.writeQuery({ query: ITEMS_QUERY, data: { items: requiredItems } })
  }
  render() {
    const { id, children } = this.props;
    return (
      <Mutation mutation={DELETE_ITEM_MUTATION}
        variables={{id}}
        update={this.update}
        >
      {(deleteItem) => {
        return (
          <button onClick={() => {
            if (confirm('Are you sure you want to delete this item')) {
              deleteItem().catch(err => alert(err));
            }
          }}>
            { children }
          </button>
        )
      }}
      </Mutation>
    )
  }
}
