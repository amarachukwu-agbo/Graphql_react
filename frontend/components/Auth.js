import React from 'react';
import { Query } from 'react-apollo';
import { CURRENT_USER_QUERY } from './User';
import SignIn from '../components/SignIn';

const Auth = ({ children }) => {
  return (
    <Query query={CURRENT_USER_QUERY}>
      {({ data , loading}) => {
        if (loading) return <p>Loading ...</p>
        if (data && !data.me) return <SignIn />
        return children;
      }}
    </Query>
  );
};

export default Auth;