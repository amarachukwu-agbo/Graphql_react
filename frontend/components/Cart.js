import React, { Component } from 'react';
import gql from 'graphql-tag';
import StripeCheckout from 'react-stripe-checkout';
import Router from 'next/router';
import NProgress from 'nprogress';
import { Query, Mutation } from 'react-apollo';
import CartStyles from './styles/CartStyles';
import SickButton from './styles/SickButton';
import CloseButton from './styles/CloseButton';
import Supreme from './styles/Supreme';
import User, { CURRENT_USER_QUERY } from './User';
import CartItem from './CartItem';
import calculateTotalPrice from '../lib/calcTotalPrice';
import formatMoney from '../lib/formatMoney';

export const LOCAL_STATE_QUERY = gql`
  query LOCAL_STATE_QUERY {
    cartOpen @client
  }
`;

export const TOGGLE_CART_MUTATION = gql`
  mutation TOGGLE_CART_MUTATION {
    toggleCart @client
  }
`;

export const CREATE_ORDER = gql`
  mutation CREATE_ORDER($token: String!) {
    createOrder(token: $token) {
      id
      items {
        image
        description
        title
      }
      total
      charge
    }
  }
`;

const totalItems = (cart) => {
  return cart.reduce((tally, item) => tally + item.quantity, 0)
}
class Cart extends Component {
  onToken = async (token, createOrder) => {
    NProgress.start();
    const order = await createOrder({
      variables: {
        token: token.id
      }
    }).catch(error => alert(`Error: ${error}`));
    Router.push({
      pathname: '/order',
      query: { id: order.data.createOrder.id }
    });
  }
  render() {
    return (
      <User>
        {({ data } ) => {
          const { me } = data;
          if (!me) return null;
          return(
            <Mutation mutation={TOGGLE_CART_MUTATION}>
              {(toggleCart) => (
              <Query query={LOCAL_STATE_QUERY}>
                {({ data }) => (
                  <CartStyles open={data.cartOpen}>
                    <header>
                      <CloseButton title="close" onClick={toggleCart}>&times;</CloseButton>
                      <Supreme>{me.name}'s Cart</Supreme>
                      <p>You have {me.cart.length} item{me.cart.length === 1 ? '': 's'} in your cart</p>
                    </header>
                    <ul>
                      { me.cart.map(item => <CartItem key={item.id} cartItem={item}/>) }
                    </ul>
                    <footer>
                      {
                        me.cart.length ? (
                          <>
                            <p>{formatMoney(calculateTotalPrice(me.cart))}</p>
                            <Mutation mutation={CREATE_ORDER} refetchQueries={[{query: CURRENT_USER_QUERY }]}>
                              {(createOrder) => (
                                <StripeCheckout
                                  name="Sick Fits"
                                  description={`Order of ${totalItems(me.cart)} items`}
                                  token={(stripeToken) => this.onToken(stripeToken, createOrder)}
                                  stripeKey="pk_test_4ApAJihRk5yShkglGcJ0ZSXV00oEwKYLu4"
                                  amount={calculateTotalPrice(me.cart)}
                                  image={me.cart.length && me.cart[0].item && me.cart[0].item.image}
                                  currency="USD"
                                >
                                  <SickButton>Checkout</SickButton>
                                </StripeCheckout>
                              )}
                            </Mutation>
                          </>
                        ) : ''
                      }
                    </footer>
                  </CartStyles>
                )}
              </Query>
              )}
            </Mutation>
          )
        }}
      </User>
    );

  }
};

export default Cart;