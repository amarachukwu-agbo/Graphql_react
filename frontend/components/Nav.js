import Link from 'next/link';
import { Mutation } from 'react-apollo';
import NavStyles from './styles/NavStyles';
import User from '../components/User';
import SignOut from '../components/SignOut';
import CartCount from '../components/CartCount';
import { TOGGLE_CART_MUTATION } from '../components/Cart';

const Nav = () => (
  <Mutation mutation={TOGGLE_CART_MUTATION}>
    {(toggleCart) => (
      <User>
        {({ data: { me }}) => (
          <NavStyles>
            <Link href="/allitems">
              <a>Shop</a>
            </Link>
            {
              me && (
                <>
                  <Link href="/sell">
                    <a>Sell!</a>
                  </Link>
                  <Link href="/orders">
                    <a>Orders</a>
                  </Link>
                  <Link href="/me">
                    <a>Account</a>
                  </Link>
                  <SignOut />
                  <button onClick={toggleCart}>
                    My Cart
                    <CartCount
                      count={me.cart.reduce((tally, cartItem) => tally + cartItem.quantity, 0)}
                    />
                  </button>
                </>
              )
            }
            {
              !me && (
                <Link href="/signup">
                  <a>Sign In</a>
                </Link>
              )
            }
      </NavStyles>
        )}
      </User>
    )}
  </Mutation>
);

export default Nav;
