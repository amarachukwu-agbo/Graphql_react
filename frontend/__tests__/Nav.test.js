import { mount } from 'enzyme';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import Nav from '../components/Nav';
import { CURRENT_USER_QUERY } from '../components/User';
import { fakeUser, fakeCartItem } from '../lib/testUtils';

const signedInMocks = [
  {
    request: {
      query: CURRENT_USER_QUERY,
    },
    result: {
      data: {
        me: fakeUser()
      }
    }
  }
];

const signedInMocksWithCartItems = [
  {
    request: {
      query: CURRENT_USER_QUERY,
    },
    result: {
      data: {
        me: {
          ...fakeUser(),
          cart: [fakeCartItem()]
        }
      }
    }
  }
];

const notSignedInMocks = [
  {
    request: {
      query: CURRENT_USER_QUERY,
    },
    result: {
      data: {
        me: null
      }
    }
  }
];

describe('<Nav />', () => {
  it('renders a minimal Nav when signed out', async () => {
    const wrapper = mount(<MockedProvider mocks={notSignedInMocks}>
      <Nav />
    </MockedProvider>);
    await wait(0);
    wrapper.update();
    expect(wrapper.find('NavStyles')).toMatchSnapshot();
  });

  it('renders the full Nav to logged in users', async () => {
    const wrapper = mount(<MockedProvider mocks={signedInMocks}>
      <Nav />
    </MockedProvider>);
    await wait(0);
    wrapper.update();
    expect(wrapper.find('NavStyles')).toMatchSnapshot();
  });

  it('renders the amount of items in cart', async () => {
    const wrapper = mount(<MockedProvider mocks={signedInMocks}>
      <Nav />
    </MockedProvider>);
    await wait(0);
    wrapper.update();
    expect(wrapper.find('NavStyles')).toMatchSnapshot();
  });
});
