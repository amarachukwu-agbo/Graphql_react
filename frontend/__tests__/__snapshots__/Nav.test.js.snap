// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> renders a minimal Nav when signed out 1`] = `
<NavStyles>
  <ul
    className="NavStyles-sc-11c0d2g-0 kpXDuX"
  >
    <Link
      href="/allitems"
    >
      <a
        href="/allitems"
        onClick={[Function]}
      >
        Shop
      </a>
    </Link>
    <Link
      href="/signup"
    >
      <a
        href="/signup"
        onClick={[Function]}
      >
        Sign In
      </a>
    </Link>
  </ul>
</NavStyles>
`;

exports[`<Nav /> renders the amount of items in cart 1`] = `
<NavStyles>
  <ul
    className="NavStyles-sc-11c0d2g-0 kpXDuX"
  >
    <Link
      href="/allitems"
    >
      <a
        href="/allitems"
        onClick={[Function]}
      >
        Shop
      </a>
    </Link>
    <Link
      href="/sell"
    >
      <a
        href="/sell"
        onClick={[Function]}
      >
        Sell!
      </a>
    </Link>
    <Link
      href="/orders"
    >
      <a
        href="/orders"
        onClick={[Function]}
      >
        Orders
      </a>
    </Link>
    <Link
      href="/me"
    >
      <a
        href="/me"
        onClick={[Function]}
      >
        Account
      </a>
    </Link>
    <SignOut>
      <Mutation
        mutation={
          Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "SIGN_OUT_MUTATION",
                },
                "operation": "mutation",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "signOut",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "message",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 70,
              "start": 0,
            },
          }
        }
        refetchQueries={
          Array [
            Object {
              "query": Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "CURRENT_USER_QUERY",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "me",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "permissions",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "orders",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "cart",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "quantity",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "item",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "id",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "price",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "description",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "image",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "title",
                                            },
                                            "selectionSet": undefined,
                                          },
                                        ],
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 270,
                  "start": 0,
                },
              },
            },
          ]
        }
      >
        <button
          onClick={[Function]}
        >
          Sign Out
        </button>
      </Mutation>
    </SignOut>
    <button
      onClick={[Function]}
    >
      My Cart
      <CartCount
        count={0}
      >
        <CartCount__AnimationStyles>
          <span
            className="CartCount__AnimationStyles-xxvp4g-1 ixnfFd"
          >
            <TransitionGroup
              childFactory={[Function]}
              component="div"
            >
              <div>
                <CSSTransition
                  className="count"
                  classNames="count"
                  in={true}
                  key=".$count"
                  onExited={[Function]}
                  timeout={
                    Object {
                      "enter": 400,
                      "exit": 400,
                    }
                  }
                  unmountOnExit={true}
                >
                  <Transition
                    appear={false}
                    className="count"
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={
                      Object {
                        "enter": 400,
                        "exit": 400,
                      }
                    }
                    unmountOnExit={true}
                  >
                    <CartCount__Count
                      className="count"
                    >
                      <div
                        className="count CartCount__Count-xxvp4g-0 lkiIUU"
                      >
                        0
                      </div>
                    </CartCount__Count>
                  </Transition>
                </CSSTransition>
              </div>
            </TransitionGroup>
          </span>
        </CartCount__AnimationStyles>
      </CartCount>
    </button>
  </ul>
</NavStyles>
`;

exports[`<Nav /> renders the full Nav to logged in users 1`] = `
<NavStyles>
  <ul
    className="NavStyles-sc-11c0d2g-0 kpXDuX"
  >
    <Link
      href="/allitems"
    >
      <a
        href="/allitems"
        onClick={[Function]}
      >
        Shop
      </a>
    </Link>
    <Link
      href="/sell"
    >
      <a
        href="/sell"
        onClick={[Function]}
      >
        Sell!
      </a>
    </Link>
    <Link
      href="/orders"
    >
      <a
        href="/orders"
        onClick={[Function]}
      >
        Orders
      </a>
    </Link>
    <Link
      href="/me"
    >
      <a
        href="/me"
        onClick={[Function]}
      >
        Account
      </a>
    </Link>
    <SignOut>
      <Mutation
        mutation={
          Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": Object {
                  "kind": "Name",
                  "value": "SIGN_OUT_MUTATION",
                },
                "operation": "mutation",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "signOut",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "message",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 70,
              "start": 0,
            },
          }
        }
        refetchQueries={
          Array [
            Object {
              "query": Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": Object {
                      "kind": "Name",
                      "value": "CURRENT_USER_QUERY",
                    },
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "me",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "email",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "name",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "permissions",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "orders",
                                },
                                "selectionSet": undefined,
                              },
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "cart",
                                },
                                "selectionSet": Object {
                                  "kind": "SelectionSet",
                                  "selections": Array [
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "id",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "quantity",
                                      },
                                      "selectionSet": undefined,
                                    },
                                    Object {
                                      "alias": undefined,
                                      "arguments": Array [],
                                      "directives": Array [],
                                      "kind": "Field",
                                      "name": Object {
                                        "kind": "Name",
                                        "value": "item",
                                      },
                                      "selectionSet": Object {
                                        "kind": "SelectionSet",
                                        "selections": Array [
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "id",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "price",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "description",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "image",
                                            },
                                            "selectionSet": undefined,
                                          },
                                          Object {
                                            "alias": undefined,
                                            "arguments": Array [],
                                            "directives": Array [],
                                            "kind": "Field",
                                            "name": Object {
                                              "kind": "Name",
                                              "value": "title",
                                            },
                                            "selectionSet": undefined,
                                          },
                                        ],
                                      },
                                    },
                                  ],
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 270,
                  "start": 0,
                },
              },
            },
          ]
        }
      >
        <button
          onClick={[Function]}
        >
          Sign Out
        </button>
      </Mutation>
    </SignOut>
    <button
      onClick={[Function]}
    >
      My Cart
      <CartCount
        count={0}
      >
        <CartCount__AnimationStyles>
          <span
            className="CartCount__AnimationStyles-xxvp4g-1 ixnfFd"
          >
            <TransitionGroup
              childFactory={[Function]}
              component="div"
            >
              <div>
                <CSSTransition
                  className="count"
                  classNames="count"
                  in={true}
                  key=".$count"
                  onExited={[Function]}
                  timeout={
                    Object {
                      "enter": 400,
                      "exit": 400,
                    }
                  }
                  unmountOnExit={true}
                >
                  <Transition
                    appear={false}
                    className="count"
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={
                      Object {
                        "enter": 400,
                        "exit": 400,
                      }
                    }
                    unmountOnExit={true}
                  >
                    <CartCount__Count
                      className="count"
                    >
                      <div
                        className="count CartCount__Count-xxvp4g-0 lkiIUU"
                      >
                        0
                      </div>
                    </CartCount__Count>
                  </Transition>
                </CSSTransition>
              </div>
            </TransitionGroup>
          </span>
        </CartCount__AnimationStyles>
      </CartCount>
    </button>
  </ul>
</NavStyles>
`;
